Assets/ARRaycast/Shaders/GridShader.shader

```diff
@@ -3,54 +3,63 @@ Shader "Custom/GridShader"
     Properties
     {
         _Color ("Color", Color) = (1,1,1,1)
-        _MainTex ("Albedo (RGB)", 2D) = "white" {}
+        _MainTex ("Texture", 2D) = "white" {}
         _Scale ("Scale", Float) = 5
     }
     SubShader
     {
         Tags { "RenderType"="Opaque" }
         LOD 200
 
-        CGPROGRAM
-        // Physically based Standard lighting model, and enable shadows on all light types
-        #pragma surface surf Standard
+        Pass
+        {
+            CGPROGRAM
+            #pragma vertex vert
+            #pragma fragment frag
 
-        // Use shader model 3.0 target, to get nicer looking lighting
-        #pragma target 3.0
+            #include "UnityCG.cginc"
 
-        sampler2D _MainTex;
+            struct appdata
+            {
+                float4 vertex : POSITION;
+                float3 normal : NORMAL;
+            };
 
-        struct Input
-        {
-            float3 worldNormal;
-            float3 worldPos;
-        };
+            struct v2f
+            {
+                float2 uv : TEXCOORD0;
+                float4 vertex : SV_POSITION;
+            };
 
-        half _Glossiness;
-        half _Metallic;
-        fixed4 _Color;
-        half _Scale;
 
-        void surf (Input IN, inout SurfaceOutputStandard o)
-        {
-            float2 uv = IN.worldPos.xy;
-            
-            if( abs(IN.worldNormal.x) > 0.707 ) uv = IN.worldPos.yz;
-            if( abs(IN.worldNormal.y) > 0.707 ) uv = IN.worldPos.xz;
-
-            uv.x *= _Scale;
-            uv.y *= _Scale;
-            
-            // Albedo comes from a texture tinted by color
-            fixed4 c = (tex2D (_MainTex, uv) + 0.05) * _Color;
-            o.Albedo = c.rgb;
-
-            // Metallic and smoothness come from slider variables
-            o.Metallic = 0;
-            o.Smoothness = 0;
-            o.Alpha = c.a;
+            sampler2D _MainTex;
+            fixed4 _Color;
+            half _Scale;
+
+            v2f vert (appdata v)
+            {
+                v2f o;
+                o.vertex = UnityObjectToClipPos(v.vertex);
+
+                // Convert the position and normal to world space for proper tiling
+                v.vertex = mul(unity_ObjectToWorld, v.vertex);
+                v.normal = mul(unity_ObjectToWorld, v.normal);
+
+                o.uv = v.vertex.xy;
```

Comment:
these 3 lines need some comments.  why is uv computed in this way?

Reply:
Adding some comments. Since this change is mostly about lighting, the uv's are being calculated the same as before.

```glsl
// Choose the axis which is most aligned with the surface normal, then use world space coordinates from
// the other two axes to drive the uvs. As a result, the texture is tiled relative to the world space.
// By default, the most-aligned normal is assumed to be the z-axis. A value > 1/sqrt(2) = 0.707 in the
// x or y value of the normal signifies a rotation > 45 degrees towards these other axes, and different
// axes should be used to calculate the uvs (yz and xz, respectively).
o.uv = v.vertex.xy;
if( abs(v.normal.x) > 0.707 ) o.uv = v.vertex.yz;
if( abs(v.normal.y) > 0.707 ) o.uv = v.vertex.xz;
````